contextual_lenses/pfam_family_classification/pfam_utils.py

```diff
@@ -80,11 +81,29 @@ def create_pfam_df(family_accessions, test=False):
   pfam_df['mod_family_accession'] = pfam_df.family_accession.apply(lambda x: mod_family_accession(x))
   pfam_df = pfam_df[pfam_df.mod_family_accession.isin(family_accessions)]
   pfam_df['index'] = pfam_df.family_id.apply(lambda x: family_id_to_index[x])
-  pfam_df['one_hot_inds'] = pfam_df.sequence.apply(lambda x: residues_to_one_hot_inds(pad_seq(x)))
+
+  pfam_df['one_hot_inds'] = pfam_df.sequence.apply(lambda x: residues_to_one_hot_inds('M' + x[:511])) # x[:512] # pad_seq(x)
 
   return pfam_df
 
 
+def create_pfam_seq_batches(family_accessions, batch_size, test=False, samples=None, epochs=1,
+                            drop_remainder=False, buffer_size=None, seed=0, random_state=0):
+  """Creates iterable object of Pfam data batches."""
+
+  pfam_df = create_pfam_df(family_accessions, test=test)
+    
+  if samples is not None:
+    pfam_df = pfam_df.sample(frac=1, replace=False, random_state=random_state)
+    pfam_df = pfam_df.groupby('mod_family_accession').head(samples).reset_index()
```

Comment:
remind me, what's mod_family_accession vs. family_accession?

Reply:
mod_family_accession just turns PFabcd.efg into PFabcd (removes decimal and things after)